doc/api/assert.md

```diff
@@ -148,6 +157,21 @@ assert.deepStrictEqual(date, fakeDate);
 // Different type tags
 ```
 
+An exception is made to the strict equality comparison rule for [`Map`][] keys
```

Comment:
This should probably go to the item 1...also the NaN case can be demonstrated by linking to the existing caveats section, something like

```
 1. For Sets and Maps, primitive values are compared using the [SameValueZero][] comparison
    (which means they are free of the [caveats][]).
    For other objects, primitive values are compared using the [Strict Equality Comparison][] ( `===` )
```